### DISCUSSION TITLE
How to use system built-in clangd?

### DISCUSSION THREAD

**Discussion by ChenZhongPu** [LazyVim/LazyVim#1351] [Q&A] [answered]
I have added the extra:

```lua
{ import = "lazyvim.plugins.extras.lang.clangd" },
```

Since my OS is shipped with `clangd`, I would like to use it instead without downloading it from mason. How to configure it while keeping all settings introduced by `lazyvim.plugins.extras.lang.clangd`?

**Comment by dpetka2001** [comment]
In your `nvim-lspconfig` spec you can define `mason = false` for `clangd`, so that it won't be installed by Mason.

**Reply by ChenZhongPu** [reply]
How to configure it? I tried:

```lua
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        clangd = { mason = false },
      },
    },
  },
```
But it is incompatible with `lazyvim.plugins.extras.lang.clangd`.

**Reply by ChenZhongPu** [reply]
Currently, `clangd` fails to work. See [#1308](https://github.com/LazyVim/LazyVim/pull/1308). I will retest it after the maintainer fix this issue.

**Reply by dpetka2001** [reply, marked as answer]
There is a solution described in #1308 for how to incorporate the changes into yours. Alternatively, you can just create a file `~/.config/nvim/lua/plugins/clangd.lua` and copy all the content of that PR into your file and it will still work. But if you do that, don't also import the extra. Just wait for it to get merged, and after that you can delete your file and import the extra again.

**Reply by naowalrahman** [reply]
Sorry to necrobump, but for anyone looking at this in the future, here is a very clean solution:
```lua
return {
    "neovim/nvim-lspconfig",
    opts = function(_, opts) 
        opts.servers.clangd.mason = false 
    end
}
```
